src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +236,81 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
+      }      
+           
+			v.setNumberOfInstances(sas.getInstances());
+			
+			return Mono.just(v);
+    });
+        
+    Flux<List<RouteResource>> routeApplicationFlux = osaVectorSpaceFlux.flatMapSequential(v -> this.getAppRoutes(v.getApplicationId()));
+		Flux<OSAVector> osaVectorApplicationRouteFlux = Flux.zip(osaVectorApplicationFlux, routeApplicationFlux).flatMap(tuple -> {
+			OSAVector v = tuple.getT1();
+			
+			if (INVALID_ROUTES == tuple.getT2()) {
+				// NB: This drops the current target!
+				return Mono.empty();
 			}
 			
-			v.setNumberOfInstances(sas.getInstances());
+      List<RouteResource> routes = tuple.getT2();
+            
+      RouteResource metricsRoute = routes.stream()
+      .filter(r -> v.getAccessURL().contains(r.getUrl()))
+      .findFirst()
+			.orElse(null);
+
+			if (metricsRoute == null) {
+				return Mono.just(v);
+			}
+
+			Destination dest = metricsRoute.getDestinations()
+			.stream()
+			.filter(d -> d.getApplication().getApplicationId().equals(v.getApplicationId()))
+			.findFirst()
+			.orElse(null);
+						
+			if (dest != null) {
+				v.setInternalPort(dest.getPort());
+				v.setRouteId(metricsRoute.getId());
+				v.setDomainId(metricsRoute.getRelationships().getDomain().getData().getId());					
+			}
+            
+      return Mono.just(v);
+    });
+    
+		Flux<GetDomainResponse> domainFlux = osaVectorApplicationRouteFlux.flatMapSequential(v -> this.cfAccessor.retrieveDomain(v.getDomainId()));
+		Flux<OSAVector> osaVectorApplicationRouteDomainFlux = Flux.zip(osaVectorApplicationRouteFlux, domainFlux).flatMap(tuple -> {
+			OSAVector v = tuple.getT1();
+			
+			if (INVALID_DOMAIN == tuple.getT2()) {
```

Comment:
see above

Reply:
I removed this as im now not sure it is required. My understanding is that flux/mono wont allow there to be a null value here anyway and if there is an error with the API call it will have thrown way before this.